daemon/api.go

```diff
@@ -1748,17 +1790,18 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 		return BadRequest("cannot use create-user as non-root")
 	}
 
-	var createData struct {
-		Email  string `json:"email"`
-		Sudoer bool   `json:"sudoer"`
-		Known  bool   `json:"known"`
-	}
+	var createData postUserCreateData
 
 	decoder := json.NewDecoder(r.Body)
 	if err := decoder.Decode(&createData); err != nil {
 		return BadRequest("cannot decode create-user data from request body: %v", err)
 	}
 
+	// special case: the user requested the creation of all known
+	// system-users
+	if createData.Email == "" && createData.Known {
+		return createAllKnownSystemUsers(c.d.overlord.State(), &createData)
```

Comment:
@niemeyer The "/v2/create-user" API now has two different return types depending on the input. One is a single JSON object describing the user (when requesting a particular user). The other is a list of JSON objects describing users (when requesting to create all user). So I think we want a new client API that has the right return type for this? Feedback on the function name for this welcome :) `CreateUsers([]string, *CreateUserRequest) ([]CreateUserResult, error)` maybe? And support that the input list can be nil? Or no input list and `CreateAllKnownUsers(req) ([]result, error)`?